### PR TITLE
{bio} [foss-2015b] bam-readcount-0.7.4 (REVIEW)

### PR DESCRIPTION
**Pull request by verdurin**
New easyconfig for bam-readcount.

The patch fixes hardcoding of the installation prefix.

### CODE REVIEW

File: easybuild/easyconfigs/b/bam-readcount/bam-readcount-0.7.4-foss-2015b.eb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+source_urls = ['https://github.com/genome/%(name)s/archive']
+
+patches = ['%(name)s-%(version)s-cmake_install_prefix.patch']
+
+builddependencies = [
+    ('SAMtools', '1.3'),
+    ('CMake', '3.4.1'),
+]
+
+configopts = " -DCMAKE_INSTALL_PREFIX=%(installdir)s "
```
Comment (boegel) [inline easybuild/easyconfigs/b/bam-readcount/bam-readcount-0.7.4-foss-2015b.eb]: this is added by default by `CMakeMake`, there should be no need for this?

File: easybuild/easyconfigs/b/bam-readcount/bam-readcount-0.7.4-cmake_install_prefix.patch [hunk 2]
```diff
@@ -0,0 +1,12 @@
+diff -u bam-readcount-0.7.4/CMakeLists.txt bam-readcount-0.7.4.new/CMakeLists.txt
```
Comment (boegel) [inline easybuild/easyconfigs/b/bam-readcount/bam-readcount-0.7.4-cmake_install_prefix.patch]: @verdurin please include a line or two of comment above, like:

```
patch out hardcoded installation prefix
author: Adam Huffman (<affiliation>)
```

(where you replace `<affiliation>`, of course :))

File: easybuild/easyconfigs/b/bam-readcount/bam-readcount-0.7.4-foss-2015b.eb (hunk 1 above)
Comment (verdurin) [inline easybuild/easyconfigs/b/bam-readcount/bam-readcount-0.7.4-foss-2015b.eb]: This was my first attempt to fix the installation problem, which was subsequently dealt with properly by the patch. Will remove.

Comment (boegel) [inline easybuild/easyconfigs/b/bam-readcount/bam-readcount-0.7.4-foss-2015b.eb]: @verdurin seems like removing this was not part of 68db97d838656d94d12133f292533dc2670c42f0 like you intended...

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
node2109.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/bc36c73f2a475412e49568d57d7038b0 for a full test report.

**Comment by boegel** [comment]
@verdurin: missing dep on `zlib`, or maybe a linking issue?

```
bam_import.c:76: error: undefined reference to 'gzopen64'
```

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 25 out of 25 (1 easyconfigs in this PR)
node2615.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/e4f7a10a4f1e4dd3a311acecbf8bb28d for a full test report.

File: easybuild/easyconfigs/b/bam-readcount/bam-readcount-0.7.4-foss-2015b.eb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+toolchain = {'name': 'foss', 'version': '2015b'}
+
+sources = ['v%(version)s.tar.gz']
+source_urls = ['https://github.com/genome/%(name)s/archive']
+
+patches = ['%(name)s-%(version)s-cmake_install_prefix.patch']
+
+builddependencies = [
+    ('SAMtools', '1.3'),
```
Comment (boegel) [inline easybuild/easyconfigs/b/bam-readcount/bam-readcount-0.7.4-foss-2015b.eb]: is listing this as a build dep correct?

also, it seems like `zlib` is missing as a direct dep (cfr. build issue on CentOS 6)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2665.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/13bec047890a7b9b32bdce31573b4d66 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2018.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/03625bafddcb946f15f87b801feb040f for a full test report.

**Comment by boegel** [comment]
lgtm

**Comment by boegel** [comment]
Going in, thanks @verdurin!

**Pull request merged**